core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -332,4 +339,26 @@ public Path tmpFile() {
     public static FileStore getFileStore(Path path) throws IOException {
         return ESFileStore.getMatchingFileStore(path, fileStores);
     }
+
+    /**
+     * assserts that the two environments are equivalent for all things the environment cares about (i.e., all but the setting
+     * object which may contain different setting)
+     */
+    public static void assertEquivalent(Environment actual, Environment expected) {
```

Comment:
It only feels natural to me to have the logic comparing two environments to be in the environment class. Makes it easier to see it's correct and that it doesn't miss anything.

> we need something stronger 

I'm open for suggestions.